Training/configs/training_v1.yaml

```diff
@@ -23,6 +23,12 @@ Setup:
     eta_min              : 0.0
     eta_max              : 2.3
     weight_thr           : 100000.0
+    # here define variables for the Histogram_2D class
+    xaxis_vec            : [0., 0.625, 1.25, 1.875, 2.5]
+    yaxis_list_vec       : [[20., 40., 60., 100., 1000.], [20., 60., 100., 1000.],
+                            [20., 100., 1000.]          , [20., 100., 1000.]     ]
+    ymin_val             : 20.0
+    ymax_val             : 1000.
 
```

Comment:
@lguzzi There might be a problem with such approach:
On practice, I think we want to save high granularity for the region of (low pt and high eta), because we have very good statistic there, and there is no reason of not to equalize the spectrum in a tight manner. So if we want to have something like this:
![image](https://user-images.githubusercontent.com/31743376/119692774-f9efd080-be4b-11eb-8d79-429fa2526d0b.png)
Will we be able to do this? because that what we actually want on practice. Or I am missing something? @kandrosov

Reply:
not exactly. What you could do is split the big bin over x and maybe use only two y bins in the last x bin

Reply:
uh... I think @shedprog is right, we've overlooked that with the current implementation the division as illustrated above is not possible... Actually, I think we can use [TH2Poly](https://root.cern.ch/doc/master/classTH2Poly.html) as an internal container in `Histogram_2D` class - it would allow us to have binning as above.

Reply:
I've updated the code and switched the x and y axes (the names of the methods and variables are the same though). 
The only modification to be done the code is now to switch x and y in the yaml file. Please note that the yaml x-binning is now the actual y binning of the input histogram, and y-binnings are now the x binnings of the input histogram; e.g. add_y_binning_by_index actually defines the binning over x for a y bin.
I've marked the code where this workaround has been introduced by a comment "// WORKAROUND" and the old code is still present and commented.
Now the code produces results like:
<img width="1006" alt="immagine" src="https://user-images.githubusercontent.com/26771212/121177463-b63e9300-c85d-11eb-823d-184ec8ea45a6.png">

Reply:
@ Is it some tesing histogram, because in our MC we have eta only up to 2.5 and here it is up to 5.0?

Reply:
yes, this is a test. The upper TH2 is actually the input for the test